lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2210,14 +2210,14 @@ def alpha_cmd(self, alpha, forced, effective_alphas):
         name = self.file.alphaState(effective_alphas)
         return [name, Op.setgstate]
 
-    def hatch_cmd(self, hatch):
+    def hatch_cmd(self, hatch, hatch_color):
```

Comment:
Also, this is made more complicated by how we implement hatching in pdf: it replaces the fill color with a named pattern. I *think* the backend_pdf changes are correct, and you added tests for the new functionality so we should be fine.